plugins/importexport/native/filter/NativeXmlSubmissionFileFilter.inc.php

```diff
@@ -163,9 +163,42 @@ function handleRevisionElement($node, $stageId, $fileId) {
 
 		$uploaderUsername = $node->getAttribute('uploader');
 		$uploaderUserGroup = $node->getAttribute('user_group_ref');
+		if (!$uploaderUsername) {
+			$uploaderUsername = $deployment->getUser()->getUsername();
+		}
+
+		// Determine the user based on the username
+		$userDao = DAORegistry::getDAO('UserDAO');
+		$user = $userDao->getByUsername($uploaderUsername);
+		if ($user) {
+			$submissionFile->setUploaderUserId($user->getId());
+		} else {
+			$deployment->addError(ASSOC_TYPE_SUBMISSION, $submission->getId(), __('plugins.importexport.common.error.unknownUploader', array('param' => $uploaderUsername)));
+			$errorOccured = true;
+		}
 
 		// Determine the user group based on the user_group_ref element.
 		$userGroupDao = DAORegistry::getDAO('UserGroupDAO');
+		if ($user && !$uploaderUserGroup) {
+			// We have a user, but no group specified in the import.  Select a default group.
+			// We will prefer any group that has access to the submission's stage; find these
+			$stageUserGroups = $userGroupDao->getUserGroupsByStage($context->getId(), $stageId);
+			$stageUserGroupLookup = array();
+			while ($userGroup = $stageUserGroups->next()) {
+				$stageUserGroupLookup[] = $userGroup->getName(null);
+			}
+			// Check all user groups from this user
+			$userUserGroups = $userGroupDao->getByUserId($user->getId(), $context->getId());
+			while ($userGroup = $userUserGroups->next()) {
+				$defaultedUserGroup = $userGroup->getName(null);
+				// If the user's user group has access to this stage, select it
+				if (in_array($defaultedUserGroup, $stageUserGroupLookup)) {
+					break;
+				}
+				// if no current user group has access to this stage, fall through with the any of the user's current user groups
+			}
+			$uploaderUserGroup = $defaultedUserGroup;
```

Comment:
@asmecher, @bozana, `UserGroup::getName()` does not acknowledge that it can take a `null` as the `$locale`, nor that it can return an `array()` (or `null`, technically) as a result.
```
	/**
	 * Get user group name
	 * @param $locale string
	 * @return string
	 */
	function getName($locale) {
		return $this->getData('name', $locale);
	}
```
If the use of `in_array($uploaderUserGroup, $userGroup->getName(null))` is legitimate, I will modify the documentation for `UserGroup::getName()`; if the use is not, I will enforce the requirement of `$locale` to `getName()` (and create a new `getNames()` function?).  What say you, oh guardians of the API?